lib/mail/yaml.rb

```diff
@@ -0,0 +1,30 @@
+require 'yaml'
+
+module Mail
+  module YAML
+    def self.load(yaml)
+      permitted_classes = [
+        Symbol,
+
+        Mail::Body,
+
+        # Delivery methods as listed in mail/configuration.rb
+        Mail::SMTP,
+        Mail::Sendmail,
+        Mail::Exim,
+        Mail::FileDelivery,
+        Mail::SMTPConnection,
+        Mail::TestMailer,
+        Mail::LoggerDelivery,
+
+        Mail.delivery_method.class,
+      ]
+
+      if Gem::Version.new(Psych::VERSION) >= Gem::Version.new('3.1.0.pre1')
```

Comment:
@mikel What is the mail support matrix? Is it worth of the version check if Ruby/Psych upstream was fine to abandon the backward compatibility?

Reply:
Hey there @voxik - I don't have a standardized one at this point mainly because with the way I enabled older ruby support it wasn't a big deal to just keep it current and mail is mainly string handling.  At some point we gotta drop 1.8 and 1.9 though :)

Reply:
Now I see that Ruby 2.5+ is mentioned on other places, which means Psych 3.0.2, so it is probably needed, unless it is reconsidered ...

Reply:
We're defining that right now 😅.

My proposal is to keep support for Ruby 2.5 and higher, and since Ruby 2.5 ships with Psych 3.0.2, I added this for backwards compatibility. I would be 100% on board to also dropping Ruby 2.5 support, and in that case I believe it would be fine to also drop support for Psych 3.0.

Whatever @mikel feels is best seems ok to me!

Reply:
I think we could match Ruby, for example https://www.ruby-lang.org/en/downloads/ states that 2.5.9 is in EOL.  So 2.5+ could be a reasonable base.

Reply:
Great, let's drop support for Ruby 2.5 and Psych 3.0 too, let me add that to this PR.

Reply:
This will probably be a Mail v3 release, because it's "very" breaking to systems running older rubies.

Reply:
BTW will dropping Ruby 2.5 support require major version bump?

Reply:
@voxik no I don't think so.  See my update of the README in this same PR for what I'm thinking.

Reply:
I suggest to at least release last 2.x version with improvements keeping 2.5 support and later (in other PR) we can prepare 3.0 release removing additional bandaids and drop 2.5 support.

Reply:
@simi I'll just release it as a the next 2.x version but drop support for rubies `< 2.5`